src/wasm/wasm.cpp

```diff
@@ -1413,6 +1413,13 @@ void Resume::finalize(Module* wasm) {
   populateResumeSentTypes(this, wasm);
 }
 
+void Suspend::finalize(Module* wasm) {
+  if (wasm) {
+    auto tag = wasm->getTag(this->tag);
+    type = tag->sig.results;
+  }
+}
+
```

Comment:
This needs to set the type to `Type::unreachable` if any of the operands are unreachable, regardless of whether `wasm` is present or not.

(We should do this for the other instructions as well if we forgot to do this before)

Reply:
I've added these checks now, here and for the earlier WasmFX instructions. If that's what you had in mind I'll update the description of the PR to mention that it also fixes finalization of the other instructions.